common/src/com/thoughtworks/go/domain/AgentInstance.java

```diff
@@ -331,6 +333,14 @@ public boolean canBeDeleted() {
         return isDisabled() && !(isBuilding() || isCancelled());
     }
 
+    public Map<String, String> getAssignmentInfo() {
```

Comment:
Why keep all of this info in a map and access it via string keys? a simple POJO should do right?